app/services/license_validation_service.rb

```diff
@@ -75,9 +75,15 @@ def validate!
 
       # Check against :user scope requirements
       if scope.present? && scope.key?(:user)
+        user_identifier = scope[:user]
+
         return [false, "user scope does not match", :USER_SCOPE_MISMATCH] unless
-          license.user&.email == scope[:user] ||
-          license.user&.id == scope[:user]
+          license.owner_id == user_identifier ||
```

Comment:
Should providing a user scope also scope any machine fingerprints to that user via `machine.owner`?

Basically, do we want user A to be able to validate machine X when machine X belongs to user B, not user A?